### PR TITLE
Add a search feature

### PR DESCRIPTION
**Pull request by danielbrzn**
Resolves https://github.com/MarkBind/markbind/issues/151

This PR adds a search feature to Markbind that can be used to traverse the titles in `site.json` and route the user to the appropriate page. It uses `typeahead.js` and `bloodhound` to achieve this.

Desktop: 
image

Mobile: 
image

It can be invoked on any page with the following HTML:

```
<div class="form-group" id="bloodhound">
  <input type="text" class="form-control typeahead" placeholder="Search">
</div>
```

The input element should have a class of `typeahead` and the enclosing div should have a class of `bloodhound`. The User Guide will be updated with these instructions.

### CODE REVIEW

File: lib/Site.js [hunk 1]
```diff
@@ -28,7 +28,6 @@ const SITE_CONFIG_DEFAULT = {
   ],
   ignore: [
     '_site/*',
-    '*.json',
```
Comment (nicholaschuayunzhi) [inline lib/Site.js:31]: It is still possible for users to ignore `.json` files and the search will no have data for the autocomplete. Maybe we can indicate in the documentation or make sure `site.json` cannot be ignored even if the users specifies for it?

File: lib/template/page.ejs [hunk 2]
```diff
@@ -17,4 +17,7 @@
 <script src="<%- asset.vue %>"></script>
 <script src="<%- asset.vueStrap %>"></script>
 <script src="<%- asset.setup %>"></script>
+<script src="<%- asset.jquery %>"></script>
+<script src="<%- asset.typeahead %>"></script>
+<script src="<%- asset.search%>"></script>
```
Comment (nicholaschuayunzhi) [inline lib/template/page.ejs:22]: Should add a space after `asset.search`

File: lib/Site.js (hunk 1 above)
Comment (acjh) [reply lib/Site.js:31]: Hmm. Authors may not want site.json to be accessible.

Comment (danielbrzn) [reply lib/Site.js:31]: The site.json is required to be accessible to provide the search results. Perhaps we can mention in the documentation that it is a requirement to have site.json to be accessible in order for search to work?

Comment (damithc) [reply lib/Site.js:31]: If the file doesn't contain confidential data such as usernames/passwords, it should be fine to expose it.

**Comment by Gisonrg** [comment]
Good job that we found the `typeahead.js` and `bloodhound` library to achieve this.
Do you think we could create a `Vue-strap` component for this? There should be some library implemented `typeahead.js` feature in Vue. The point I want to make is try not import the jquery dependency again as we already have jquery [installed](https://github.com/MarkBind/vue-strap/blob/master/package.json#L16) in the generated `Vue-strap` bundle (it is exported as `jQuery` in generated site if you type that in your console). 
This seems a quite frontend change, by creating a new component with proper attributes taken in make senes to me. It would also be easier for authors to use as well, I can image something like `<search></search>` or similar. You can define the default search behaviour for this component. For example, starting with a component that will search the `site.json` by default then improve it as more requirements coming in.

**Comment by rachx** [comment]
There is already https://yuche.github.io/vue-strap/#typeahead this in the original vue-strap. Not sure if we can use it.

**Comment by danielbrzn** [comment]
There's `Typeahead.vue` that's already existing in Vue-strap. I'll figure out a way to integrate search with that instead of relying on `typeahead.js` and `bloodhound`.

**Pull request closed without merging**